src/app/clusters/general-diagnostics-server/general-diagnostics-server.cpp

```diff
@@ -290,6 +323,51 @@ GeneralDiagnosticsDelegate gDiagnosticDelegate;
 
 } // anonymous namespace
 
+bool emberAfGeneralDiagnosticsClusterTestEventTriggerCallback(CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
+                                                              const Commands::TestEventTrigger::DecodableType & commandData)
+{
+
+    if (commandData.enableKey.size() != TestEventTriggerDelegate::kExpectedEnableKeyLength)
+    {
+        commandObj->AddStatus(commandPath, Status::ConstraintError);
+        return true;
+    }
+
+    if (IsByteSpanAllZeros(commandData.enableKey))
+    {
+        commandObj->AddStatus(commandPath, Status::ConstraintError);
+        return true;
+    }
+
+    auto * testEventTrigger = Server::GetInstance().GetTestEventTriggerDelegate();
+    if (testEventTrigger == nullptr)
+    {
+        commandObj->AddStatus(commandPath, Status::InvalidCommand);
```

Comment:
This is equivalent to the key not matching (it could never match, because we don't even have a handler). This should return `UNSUPPORTED_ACCESS`